protostar/cli/network_command_mixin.py

```diff
@@ -0,0 +1,80 @@
+from logging import Logger
+from typing import Any
+
+from starknet_py.net.gateway_client import GatewayClient
+
+from protostar.cli import Command
+from protostar.protostar_exception import ProtostarException
+from protostar.starknet_gateway import NetworkConfig
+from protostar.starknet_gateway.network_config import is_legacy_network_name, NETWORKS
+
+GATEWAY_URL_ARG_NAME = "gateway-url"
+NETWORK_ARG_NAME = "network"
+CHAIN_ID_ARG_NAME = "chain-id"
+
+
+class NetworkCommandMixin:
+    @property
+    def network_arguments(self):
+        return [
+            Command.Argument(
+                name=GATEWAY_URL_ARG_NAME,
+                description="The URL of a StarkNet gateway. It is required unless `--network` is provided.",
+                type="str",
+            ),
+            Command.Argument(
+                name=CHAIN_ID_ARG_NAME,
+                description="The chain id. It is required unless `--network` is provided.",
+                type="int",
+            ),
+            Command.Argument(
+                name=NETWORK_ARG_NAME,
+                short_name="n",
+                description=(
+                    "\n".join(
+                        [
+                            "The name of the StarkNet network.",
+                            "It is required unless `--gateway-url` is provided.",
+                            "",
+                            "Supported StarkNet networks:",
+                        ]
+                        + [f"- `{n}`" for n in NETWORKS]
+                    )
+                ),
+                type="str",
+            ),
+        ]
+
+    @staticmethod
+    def validate_network_command_args(args, logger: Logger):
+        if args.network is None and args.gateway_url is None:
+            raise ProtostarException(
+                f"Argument `{GATEWAY_URL_ARG_NAME}` or `{NETWORK_ARG_NAME}` is required"
+            )
+
+        # TODO(arcticae): Remove in the future version, with the legacy names formats support
+        if args.network and is_legacy_network_name(args.network):
+            logger.warning(
+                f"{args.network} is a legacy network name parameter and it won't be supported in future versions"
+            )
+
+        if args.gateway_url and not args.chain_id:
+            raise ProtostarException(
+                f"Argument `{CHAIN_ID_ARG_NAME}` is required when `{GATEWAY_URL_ARG_NAME}` is provided"
+            )
```

Comment:
it's a dependency of GatewayClient

Reply:
BTW Not anymore in 0.10 for starknet.py - so that's going to get required only for default signer usage scenario

Reply:
Isn't that going to be removed soon? I heard that the same chain_id is used by devnet and testnet so that could be the default value.

Reply:
Yeah it'd be reasonable to provide a default one, but i suppose it could lead to signatures being incorrect. It could be hard to trace back though, since there's no direct mechanism for verifying this parameter in the contract, only the whole signature